services/QuillLMS/app/services/classlink_integration/user_persister.rb

```diff
@@ -12,6 +14,7 @@ def initialize(incoming_user:, existing_provider_users:, classlink_instance_id:)
 
     def run
       creatable_user_hash = UserPersister.creatable_user(incoming_user)
+      raise ValidationError unless self.class.valid?(creatable_user_hash)
```

Comment:
The `valid?` code feels like something that should be asked of user.rb:

`def self.safe_role?(role) = SAFE_ROLES.include?(role)`

Then your you can get rid of the `self.valid?` and make the check directly:

`raise ValidationError unless User.safe_role?(creatable_user_hash[:role])`